app-backend/api/src/main/scala/campaign/Routes.scala

```diff
@@ -0,0 +1,250 @@
+package com.rasterfoundry.api.campaign
+
+import com.rasterfoundry.akkautil._
+import com.rasterfoundry.api.utils.queryparams.QueryParametersCommon
+import com.rasterfoundry.database._
+import com.rasterfoundry.datamodel._
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server._
+import cats.effect.IO
+import cats.implicits._
+import de.heikoseeberger.akkahttpcirce.ErrorAccumulatingCirceSupport._
+import doobie._
+import doobie.implicits._
+import doobie.util.transactor.Transactor
+
+import java.util.UUID
+
+trait CampaignRoutes
+    extends CommonHandlers
+    with Directives
+    with Authentication
+    with PaginationDirectives
+    with QueryParametersCommon
+    with CampaignProjectRoutes
+    with CampaignPermissionRoutes {
+
+  val xa: Transactor[IO]
+
+  val campaignRoutes: Route = {
+    pathEndOrSingleSlash {
+      get {
+        listCampaigns
+      } ~
+        post {
+          createCampaign
+        }
+    } ~
+      pathPrefix(JavaUUID) { campaignId =>
+        pathEndOrSingleSlash {
+          get {
+            getCampaign(campaignId)
+          } ~
+            put {
+              updateCampaign(campaignId)
+            } ~
+            delete {
+              deleteCampaign(campaignId)
+            }
+        } ~
+          pathPrefix("permissions") {
+            pathEndOrSingleSlash {
+              get {
+                listCampaignPermissions(campaignId)
+              } ~
+                put {
+                  replaceCampaignPermissions(campaignId)
+                } ~
+                post {
+                  addCampaignPermission(campaignId)
+                } ~
+                delete {
+                  deleteCampaignPermissions(campaignId)
+                }
+            }
+          } ~
+          pathPrefix("projects") {
+            pathEndOrSingleSlash {
+              get {
+                listCampaignProjects(campaignId)
+              }
+            }
+          } ~ pathPrefix("actions") {
+          pathEndOrSingleSlash {
+            get {
+              listCampaignUserActions(campaignId)
+            }
+          }
+        }
+      }
+  }
+
+  def listCampaigns: Route = authenticate { user =>
+    authorizeScope(
+      ScopedAction(Domain.Campaigns, Action.Read, None),
+      user
+    ) {
+      (withPagination & campaignQueryParameters) { (page, campaignQP) =>
+        complete {
+          CampaignDao
+            .listCampaigns(page, campaignQP, user)
+            .transact(xa)
+            .unsafeToFuture
+        }
+      }
+    }
+  }
+
+  def createCampaign: Route = authenticate { user =>
+    authorizeScopeLimit(
+      CampaignDao.countUserCampaigns(user).transact(xa).unsafeToFuture,
+      Domain.Campaigns,
+      Action.Create,
+      user
+    ) {
+      entity(as[Campaign.Create]) { newCampaign =>
+        onSuccess(
+          CampaignDao
+            .insertCampaign(newCampaign, user)
```

Comment:
for posterity, this is where `campaign` stopped looking like a word to me, and the scrollbar is about 15% of the way through :fearful: